pkg/sql/opt/exec/execbuilder/tests/local/local_test: TestExecBuild_partial_stats failed

pkg/sql/opt/exec/execbuilder/tests/local/local_test.TestExecBuild_partial_stats [failed](https://mesolite.cluster.engflow.com/invocations/default/ff4dd15d-cd4f-441a-a639-17ec6b6c5df3?testReportRun=9&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cvc3FsL29wdC9leGVjL2V4ZWNidWlsZGVyL3Rlc3RzL2xvY2FsOmxvY2FsX3Rlc3Q=) on release-24.3 @ [82e3e3e59011fc3e4317362fd1204caf42f8b0bd](https://github.com/cockroachdb/cockroach/commits/82e3e3e59011fc3e4317362fd1204caf42f8b0bd):


```
=== RUN   TestExecBuild_partial_stats
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestExecBuild_partial_stats525738773
    test_log_scope.go:76: use -show-logs to present logs inline
    logic.go:3109: let $forecastsEnabledPrev = false
    logic.go:3109: let $hist_id_ka_outer_buckets_full = 1085517456385867777
    logic.go:3063: 
         
        /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/partial_stats:1402: EXPLAIN SELECT * FROM ka WHERE a > 5
        expected:
            distribution: local
            vectorized: true
            ·
            • scan
              estimated row count: 5 (45% of the table; stats collected <hidden> ago)
              table: ka@ka_a_idx
              spans: [/6 - ]
            
        but found (query options: "") :
            distribution: local
            vectorized: true
            ·
            • scan
              estimated row count: 4 (40% of the table; stats collected <hidden> ago)
              table: ka@ka_a_idx
              spans: [/6 - ]
            
[04:27:18] --- done: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/partial_stats with config local: 55 tests, 1 failures
    logic.go:4249: 
        /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/partial_stats:1413: error while processing
    logic.go:4249: /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/partial_stats:1413: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestExecBuild_partial_stats525738773
--- FAIL: TestExecBuild_partial_stats (2.20s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=9</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_partial_stats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-52083

--- a/pkg/sql/opt/exec/execbuilder/testdata/partial_stats
+++ b/pkg/sql/opt/exec/execbuilder/testdata/partial_stats
@@ -1390,7 +1390,12 @@ INSERT INTO ka VALUES (10, 10)
 statement ok
 CREATE STATISTICS ka_partialstat ON a FROM ka USING EXTREMES
 
-query T retry
+# Now clear the stats cache so that the query below is guaranteed to pick up the
+# new stats (partial and merged).
+statement ok
+SELECT crdb_internal.clear_table_stats_cache();
+
+query T
 EXPLAIN SELECT * FROM ka WHERE a > 5
 ----
 distribution: local

